docker/trigger_lint_on_native_content.py

```diff
@@ -0,0 +1,147 @@
+parser.add_argument('-ti', '--target-image', help='The image to run lint with', required=True)
+parser.add_argument('-ct', '--circle-token', help='The token to trigger circle pipelines with', required=True)
+parser.add_argument('-dd', '--docker-dirs', help='The docker dirs that were changed', required=True)
+
+
+def trigger_pipeline(target_image: str, circle_token: str):
+    payload = f'{{\"branch\":\"master\",' \
+              f'\"parameters\":{{\"docker_image_target\":\"{target_image}\"}}}}'
+
+    headers = {
+        'Content-Type': "application/json",
+        'Circle-Token': circle_token
+    }
+
+    res = requests.post(f"{CIRCLE_ADDRESS}/{API_PROJECT_URL}/pipeline", data=payload, headers=headers)
+
+    data = res.text
+
+    try:
+        pipeline_data = json.loads(data)
+    except Exception as err:
+        logging.error(f'Could not parse circle response: {str(err)}')
+        logging.error(f'\nRaw Response:\n {data}')
+        raise err
+
+    pipeline_number = pipeline_data.get("number")
+    pipeline_id = pipeline_data.get("id")
+
+    if not pipeline_number:
+        raise Exception(f"Could not trigger pipeline: {pipeline_data.get('message')}")
+
+    return pipeline_number, pipeline_id
+
+
+def get_pipeline_state(pipeline_num: str, circle_token: str):
+    headers = {
+        'Circle-Token': circle_token
+    }
+
+    res = requests.get(f"{CIRCLE_ADDRESS}/{API_PROJECT_URL}/pipeline/{pipeline_num}", headers=headers)
+
+    data = res.text
+
+    try:
+        pipeline_data = json.loads(data)
+    except Exception as err:
+        logging.error(f'Could not parse circle status response: {str(err)}')
+        raise err
+
+    pipeline_state = pipeline_data.get("state")
+
+    if not pipeline_state:
+        raise Exception(f"Could not poll pipeline status: {pipeline_data.get('message')}")
+
+    return pipeline_state
+
+
+def get_pipeline_workflow_status(pipeline_id: str, circle_token: str):
+    headers = {
+        'Circle-Token': circle_token
+    }
+
+    res = requests.get(f"{CIRCLE_ADDRESS}/api/v2/pipeline/{pipeline_id}/workflow", headers=headers)
+
+    data = res.text
+
+    try:
+        pipeline_data = json.loads(data)
+    except Exception as err:
+        logging.error(f'Could not parse circle status response: {str(err)}')
+        raise err
+
+    workflow_state = None
+    for workflow in pipeline_data.get('items'):
+        if workflow.get('name') == 'native_image_lint_trigger':
+            workflow_state = workflow.get('status')
+
+    if not workflow_state:
+        raise Exception(f"Could not poll pipeline status: {pipeline_data}")
+
+    return workflow_state
+
+
+def main(target_image: str, circle_token: str, docker_dirs: str):
+    logging.info(f'Checking if changed dockerfiles are of native image')
+    docker_dirs_list = docker_dirs.split("\n")
+    logging.info(f'Changed docker dirs found: {docker_dirs_list}')
+    if NATIVE_IMAGE_DIR_NAME not in docker_dirs_list:
```

Comment:
this is a redundant check. the only thing that should be checked here is the native image, so `if [[ $(cat "/tmp/image_full_name.txt")` from the workflow script should be enough